src/emulator/storage/apis/firebase.ts

```diff
@@ -104,7 +110,12 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
       if (err instanceof NotFoundError) {
         return res.sendStatus(404);
       } else if (err instanceof ForbiddenError) {
-        return res.sendStatus(403);
+        return res.status(403).json({
+          error: {
+            code: 403,
+            message: `Permission denied. No READ permission.`,
+          },
+        });
       }
       throw err;
```

Comment:
Not entirely familiar of the standard for when we're throwing an error vs returning a status code, but could we throw a `400` error as the catch all case?

Reply:
The intent of throwing an error here is to signal there is something fundamentally wrong with the emulator, ie an unexpected error case. Returning a 400 Bad Request can be confusing to users as they would assume something wrong with their request as opposed to something broken in the emulator. A better error code to return would be a 500 Internal, but raising an error and letting the emulator die is easier to debug than trying to understand why the "server" is failing

Reply:
I see makes sense, thanks for the info!